src/workers/evaluation_environment.rs

```diff
@@ -0,0 +1,333 @@
+
+        let eval_ctx = self.policy_id_to_eval_ctx.get(policy_id).ok_or(anyhow!(
+            "cannot find evaluation context for policy with ID {policy_id}"
+        ))?;
+
+        Ok(evaluator.validate(req.req.clone(), &settings.settings, eval_ctx))
+    }
+
+    /// Validate the settings the user provided for the given policy
+    pub fn validate_settings(&self, policy_id: &str) -> Result<SettingsValidationResponse> {
+        let settings = self.policy_id_to_settings.get(policy_id).ok_or(anyhow!(
+            "cannot find settings for policy with ID {policy_id}"
+        ))?;
+
+        let mut evaluator = self.rehydrate(policy_id)?;
+
+        Ok(evaluator.validate_settings(&settings.settings))
+    }
+
+    /// Internal method, create a `PolicyEvaluator` by using a pre-initialized instance
+    fn rehydrate(&self, policy_id: &str) -> Result<PolicyEvaluator> {
+        let module_digest = self
+            .policy_id_to_module_digest
+            .get(policy_id)
+            .ok_or(anyhow!(
+                "cannot find module_digest for policy with ID {policy_id}"
+            ))?;
+        let policy_evaluator_pre = self
+            .module_digest_to_policy_evaluator_pre
+            .get(module_digest)
+            .ok_or(anyhow!(
+                "cannot find PolicyEvaluatorPre for policy with ID {policy_id}"
+            ))?;
+
+        let eval_ctx = self.policy_id_to_eval_ctx.get(policy_id).ok_or(anyhow!(
+            "cannot find evaluation context for policy with ID {policy_id}"
+        ))?;
+
+        policy_evaluator_pre.rehydrate(eval_ctx)
+    }
+}
+
+fn create_wasmtime_module(
+    policy_url: &str,
+    engine: &wasmtime::Engine,
+    precompiled_policy: &PrecompiledPolicy,
+) -> Result<wasmtime::Module> {
+    // See `wasmtime::Module::deserialize` to know why this method is `unsafe`.
+    // However, in our context, nothing bad will happen because we have
+    // full control of the precompiled module. This is generated by the
+    // WorkerPool thread
+    unsafe { wasmtime::Module::deserialize(engine, &precompiled_policy.precompiled_module) }
+        .map_err(|e| anyhow!("could not rehydrate wasmtime::Module {policy_url}: {e:?}"))
+}
+
+/// Internal function, takes care of creating the `PolicyEvaluator` instance for the given policy
+fn create_policy_evaluator_pre(
+    engine: &wasmtime::Engine,
+    module: &wasmtime::Module,
+    mode: PolicyExecutionMode,
+    policy_evaluation_limit_seconds: Option<u64>,
+) -> Result<PolicyEvaluatorPre> {
+    let mut policy_evaluator_builder = PolicyEvaluatorBuilder::new()
+        .engine(engine.to_owned())
+        .policy_module(module.to_owned())
+        .execution_mode(mode);
+
+    if let Some(limit) = policy_evaluation_limit_seconds {
+        policy_evaluator_builder =
+            policy_evaluator_builder.enable_epoch_interruptions(limit, limit);
+    }
+
+    policy_evaluator_builder.build_pre()
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
maybe we should add more unit tests here if possible,  for instance we could test the public methods that do not involve an evaluation.

Reply:
On it :+1:

Reply:
Done